OutOfSchool/OutOfSchool.IdentityServer/Services/ProviderAdminChangesLogService.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using OutOfSchool.IdentityServer.Services.Interfaces;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+
+namespace OutOfSchool.IdentityServer.Services
+{
+    public class ProviderAdminChangesLogService : IProviderAdminChangesLogService
+    {
+        private readonly IEntityRepository<ProviderAdminChangesLog> providerAdminChangesLogRepository;
+
+        public ProviderAdminChangesLogService(IEntityRepository<ProviderAdminChangesLog> providerAdminChangesLogRepository)
+        {
+            this.providerAdminChangesLogRepository = providerAdminChangesLogRepository;
+        }
+
+        public async Task<int> SaveChangesLogAsync(ProviderAdmin entity, string userId, OperationType operationType)
+        {
```

Comment:
Check `ProviderAdmin entity` on null

Reply:
Given service usage scenarios, we should never have nulls when calling this method. I would leave it as is.

Reply:
Yeah, but you know this. What will happen if I don't know and my entity will be null because some methods allows null? Your method will throw null reference that is much worse for understanding that argument null exception. You cannot predict how somebody uses that considering its `public` modifier which is accesable outside of other projects.